code/modules/cargo/bounties/reagent.dm

```diff
@@ -0,0 +1,123 @@
+/datum/bounty/reagent
+	var/required_volume = 10
+	var/shipped_volume = 0
+	var/datum/reagent/wanted_reagent
+
+/datum/bounty/reagent/completion_string()
+	return {"[round(shipped_volume)]/[required_volume] Units"}
+
+/datum/bounty/reagent/can_claim()
+	return ..() && shipped_volume >= required_volume
+
+/datum/bounty/reagent/applies_to(obj/O)
+	if(!istype(O, /obj/item/weapon/reagent_containers))
+		return FALSE
+	if(!O.reagents || !O.reagents.has_reagent(wanted_reagent.id))
+		return FALSE
+	return shipped_volume < required_volume
+
+/datum/bounty/reagent/ship(obj/O)
+	if(!applies_to(O))
+		return
+	shipped_volume += O.reagents.get_reagent_amount(wanted_reagent.id)
+	if(shipped_volume > required_volume)
+		shipped_volume = required_volume
+
+/datum/bounty/reagent/compatible_with(other_bounty)
+	if(!istype(other_bounty, /datum/bounty/reagent))
+		return TRUE
+	var/datum/bounty/reagent/R = other_bounty
+	return wanted_reagent.id != R.wanted_reagent.id
+
+/datum/bounty/reagent/simple_drink
+	name = "Simple Drink"
+	reward = 1500
+
+datum/bounty/reagent/simple_drink/New()
+	// Don't worry about making this comprehensive. It doesn't matter if some drinks are skipped.
+	var/static/list/possible_reagents = list(\
```

Comment:
This list generation will need to be better. I think it would be wise to make a thing that includes all types of ethanol based drinks. Same goes for all the other bounties that require other reagents.

Reply:
I dont agree with that, as you don't want to pay a bounty of 1500 / 4000 credits for something that someone can just pour out of a bottle.

But feel free to send me some drinks that should be added via dm